src/preferences/PreferencesBase.js

```diff
@@ -144,13 +145,15 @@ define(function (require, exports, module) {
      * 
      * @constructor
      * @param {string} path Path to the preferences file
-     * @param {boolean} createIfNew True if the file should be created if it doesn't exist.
+     * @param {boolean} createIfMissing True if the file should be created if it doesn't exist.
      *                              If this is not true, an exception will be thrown if the
      *                              file does not exist.
+     * @param {boolean} isStateJson True if the specified preferences file is state.json 
      */
-    function FileStorage(path, createIfNew) {
+    function FileStorage(path, createIfMissing, isStateJson) {
         this.path = path;
-        this.createIfNew = createIfNew;
+        this.createIfMissing = createIfMissing;
+        this.isStateJson = isStateJson;
```

Comment:
@prksingh @rroshan1  Please rename this variable to something more generic, not `isStateJson`. If we have a similar requirement in the future we could use this flag, and in that case `isStateJson` does not make sense.